### ISSUE TITLE
Exclude Codegen files from linting

### ISSUE THREAD

**Issue by kacper1703** [realm/SwiftLint#2832]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint causes errors in Xcode-Generated files like SiriKit Intents. These files cannot be added to the project as they are in the derived data folder.

### Environment

* SwiftLint version: 0.34.0
* Installation method: CocoaPods
* Configuration file:

```yml
line_length:
  warning: 160
  error: 200
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **No**
* Xcode version: 10.2.1 (10E1001)

### Result

SwiftLint raises errors in Xcode-generated files. Here an example of a definition raising the `line_length` error:
```swift
    /*!
     @abstract Initializes the response object with the specified code and user activity object.
     @discussion The app extension has the option of capturing its private state as an NSUserActivity and returning it as the 'currentActivity'. If the app is launched, an NSUserActivity will be passed in with the private state. The NSUserActivity may also be used to query the app's UI extension (if provided) for a view controller representing the current intent handling state. In the case of app launch, the NSUserActivity will have its activityType set to the name of the intent. This intent object will also be available in the NSUserActivity.interaction property.

     @param  code The response code indicating your success or failure in confirming or handling the intent.
     @param  userActivity The user activity object to use when launching your app. Provide an object if you want to add information that is specific to your app. If you specify nil, the system automatically creates a user activity object for you, sets its type to the class name of the intent being handled, and fills it with an INInteraction object containing the intent and your response.
     */
    @objc(initWithCode:userActivity:)
    public convenience init(code: IntentResponseCode, userActivity: NSUserActivity?) {
        self.init()
        self.code = code
        self.userActivity = userActivity
    }
```

To reproduce, create a project, add a Siri Extension, in `.intentsdefinition` file add a custom intent and build.

**Comment by kacper1703** [comment]
I also tried excluding the files using a regex like :
```yaml
exclude: 
  - "*.Intent.swift"
```

But it doesn't work.

**Comment by kacper1703** [comment]
Ok, it was an issue on my side. Gradle generated some leftover files and I had to remove them.
